src/test/java/in/projecteka/consentmanager/user/PatientControllerTest.java

```diff
@@ -274,4 +269,105 @@ public void shouldReturnErrorWhenPwdIsInvalidWhileUpdating() {
+                .newPassword("Test@2020")
+                .build();
+        String userName = "user@ncg";
+        Session expectedSession = Session.builder()
+                .accessToken("New access token")
+                .tokenType("bearer")
+                .build();
+        var token = string();
+
+        when(authenticator.verify(token)).thenReturn(Mono.just(new Caller(userName, true)));
+        when(userService.updatePasswordFor(request, userName)).thenReturn(Mono.just(expectedSession));
+
+        webClient.put()
+                .uri("/patients/profile/update-password")
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(AUTHORIZATION, token)
+                .accept(MediaType.APPLICATION_JSON)
+                .bodyValue(request)
+                .exchange()
+                .expectStatus()
+                .isOk();
+
+        verify(userService, times(1)).updatePasswordFor(request, userName);
+    }
+
+    @Test
+    public void shouldReturnErrorForInvalidPasswordUpdateRequest() {
+        UpdatePasswordRequest request = UpdatePasswordRequest.builder()
+                .oldPassword("Test@1234")
+                .newPassword("Test")
+                .build();
+        String userName = "user@ncg";
+        Session expectedSession = Session.builder()
+                .accessToken("New access token")
+                .tokenType("bearer")
+                .build();
+        var token = string();
+
+        when(authenticator.verify(token)).thenReturn(Mono.just(new Caller(userName, true)));
+
+        webClient.put()
+                .uri("/patients/profile/update-password")
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(AUTHORIZATION, token)
+                .accept(MediaType.APPLICATION_JSON)
+                .bodyValue(request)
+                .exchange()
+                .expectStatus()
+                .isBadRequest();
+    }
+
+    @Test
+    public void shouldReturnErrorOnUpdatePasswordFails() {
+        UpdatePasswordRequest request = UpdatePasswordRequest.builder()
+                .oldPassword("Test@1234")
+                .newPassword("TestPassword@2020")
+                .build();
+        String userName = "user@ncg";
+        var token = string();
+
+        when(authenticator.verify(token)).thenReturn(Mono.just(new Caller(userName, true)));
+        when(userService.updatePasswordFor(request, userName)).thenReturn(Mono.error(ClientError.failedToUpdatePassword()));
+
+        webClient.put()
+                .uri("/patients/profile/update-password")
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(AUTHORIZATION, token)
+                .accept(MediaType.APPLICATION_JSON)
+                .bodyValue(request)
+                .exchange()
+                .expectStatus()
+                .is5xxServerError();
+    }
+
+    @Test
+    public void shouldReturnErrorIfOldPasswordValidationFails() {
```

Comment:
tests the case where the oldpassword given is incorrect.

Reply:
Updated